geoportal/c2cgeoportal_geoportal/lib/functionality.py

```diff
@@ -95,22 +104,32 @@ def get_functionality(name, request):
 
     if request.user is not None:
         result = _get_db_functionality(
-            name, request.user, FUNCTIONALITIES_TYPES, request, errors
+            name, _user_to_struct(request.user), FUNCTIONALITIES_TYPES, request, errors
         )
+        if len(result) == 0:
+            result = _get_db_functionality(
+                name, _get_user('registered'), FUNCTIONALITIES_TYPES, request, errors
+            )
+
+    if len(result) == 0 and is_intranet:
+        result = _get_db_functionality(
+            name, _get_user('intranet'), FUNCTIONALITIES_TYPES, request, errors
+        )
+
     if len(result) == 0:
-        result = _get_config_functionality(
-            name, request.user is not None, FUNCTIONALITIES_TYPES, request, errors
+        result = _get_db_functionality(
+            name, _get_user('anonymous'), FUNCTIONALITIES_TYPES, request, errors
         )
 
     if errors != set():  # pragma: no cover
-        log.error("\n".join(errors))
+        LOG.error("\n".join(errors))
```

Comment:
Is it possible to do this on the whole code in a dedicated pull request.

Reply:
+1